Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptIDLookup.java

```diff
@@ -84,6 +90,19 @@
 
         if (entry != null) {
 
+            try {
+                // Check if the id used in a generic id. If so fetch the concept
+                // wrapper id for the generic wordnet id
+                if (conceptTypesService
+                        .getConceptTypeByConceptId(pathParts[lastIndex]) == ConceptTypes.GENERIC_WORDNET_CONCEPT) {
+                    ConceptEntry wrapperConceptEntry = conceptManager.getConceptEntry(entry.getId());
+                    entry.setId(wrapperConceptEntry.getId());
```

Comment:
I think you need to use the wordnet wrapping concept for everything, not just the id. The description could be different too and you  want the type etc. These things won't be on the entry you get when querying for the general wordnet id.